documentation/cadl-structure-guidelines.md

```diff
@@ -0,0 +1,157 @@
+
+## Service Family Libraries
+
+The service family library concept allows a family of services to share common models, linter rules, templates, etc.
+
+Service libraries can include unpublished service _family_ libraries via source dependency annotations in `package.json`:
+
+```json
+"peerDependencies": {
+  "@azure-tools/cadl-azure-core": "~x.x.x",
+  "@cadl-api-spec/azure-communication-common": "file: ../Communication.Shared"
+},
+```
+
+While this would permit services from importing any service library described in the specs repo, as a matter of policy we should probably avoid that and have tooling to detect this scenario. Service family libraries **should** use versioning decorators and spec packages should reference them as versioned dependencies. Tooling would need to ensure that changes to service family libraries does not result in unexpected changes to any service version. One way to do this would be to diff the projection of the service versions on the `main` branch against the projection of the service versions that result from the change.
+
+We treat the shared library as a sibling with other packages within the service family. This is similar to what we currently do for services that have a "Shared" package and would allow an arbitrary number of shared packages.
+
+```
+-> specification
+   -> communication
+      -> Communication.Chat         (data-plane)
+      -> Communication.Calling      (data-plane)
+      -> Communication.Management   (management)
+      -> Communication.Shared       (shared)
+```
+
+Here's an example of how Cognitive Services might use multiple shared libraries:
+
+```
+-> specification
+   -> cognitiveservices
+      -> Language.TextAnalytics  (data-plane)
+      -> Language.QnA            (data-plane)
+      -> Language.Shared         (shared)
+      -> Vision.ComputerVision   (data-plane)
+      -> Vision.CustomVision     (data-plane)
+      -> Vision.Shared           (shared)
+```
+
+# Spec Versioning
+
+Cadl has a versioning library which allows a single spec to represent multiple versions through projections. Service teams have GA service versions as well as public and private Preview service versions. The versioning library is currently optimized for the kinds of changes allowed between GA service releases: long-lived, stable, backward compatible changes. Preview versions are shorter-lived and often have wildly breaking changes from one version to the next, for which the versioning library is not optimized.
+
+### Goals
+
+Our versioning story should account for the following key goals:
+
+- It should require minimal effort for a service team to convert a preview spec to a GA one.
+- It should be easy to retire a preview spec.
+- It should be able to model all supported preview versions.
+- It must be possible to fork a spec that has breaking changes that exceed the expressiveness of the versioning library decorators.
```

Comment:
This is merely the "goals" of the versioning section. None of these bullet points are prescriptive. I could actually remove them since I'm not sure they are relevant to a general audience.